packages/config-service/src/services/globalConfig.ts

```diff
@@ -11,19 +11,22 @@
 type ExtractTypeStringFromKey<K extends string> = K extends keyof typeof _CONFIG ? (typeof _CONFIG)[K]['type'] : never;
 
 /**
- * Maps string representations of types (`'string'`, `'boolean'`, `'number'`) to their actual TypeScript types.
+ * Maps string representations of types (`'string'`, `'boolean'`, `'number'`, `'array'`) to their actual TypeScript types.
  *
  * Example:
  * - `'string'` → string
  * - `'boolean'` → boolean
  * - `'number'` → number
+ * - `'array'` → any[]
  */
 type StringTypeToActualType<Tstr extends string> = Tstr extends 'string'
   ? string
   : Tstr extends 'boolean'
   ? boolean
   : Tstr extends 'number'
   ? number
+  : Tstr extends 'array'
+  ? any[]
```

Comment:
should we refine this to `string[]|number[]`? this is to avoid `any` and to be more declarative.

Or should we even have two different array types, `'num-array'` and `'str-array'` to distinguish between them.

Reply:
Yes, I would go with two different types as it's more precise